src/core/ext/transport/chttp2/transport/chttp2_transport.cc

```diff
@@ -1409,21 +1407,13 @@ static void perform_stream_op_locked(void* stream_op,
   }
 
   grpc_closure* on_complete = op->on_complete;
-  // TODO(roth): This is a hack needed because we use data inside of the
-  // closure itself to do the barrier calculation (i.e., to ensure that
-  // we don't schedule the closure until all ops in the batch have been
-  // completed).  This can go away once we move to a new C++ closure API
-  // that provides the ability to create a barrier closure.
-  if (on_complete == nullptr) {
-    on_complete = GRPC_CLOSURE_INIT(&op->handler_private.closure, do_nothing,
-                                    nullptr, grpc_schedule_on_exec_ctx);
+  if (on_complete != nullptr) {
+    /* This batch has send ops. Use final_data as a barrier until enqueue time;
```

Comment:
Can you switch to C++ comments for consistency

Reply:
Can you also add a comment above the if statement explaining the circumstances under which on_complete would either be nullptr or not

Reply:
done!